### PR TITLE
LW-10137 onboarding tech debt

### PR DESCRIPTION
**Pull request by emiride**
# Checklist

- [x] JIRA - LW-10137

---

## Proposed solution

Remove legacy code, such that there is no duplication and there are no unused functions or classes.

## Testing

Ran the regression from branch to ensure nothing else is broken.
https://0a4e-185-161-93-8.ngrok-free.app/index.html

### CODE REVIEW

File: packages/e2e-tests/src/features/analytics/AnalyticsForgotPassword.feature [hunk 1]
```diff
@@ -50,7 +50,7 @@ Feature: Analytics - Forgot Password
     Then I validate latest analytics single event "unlock wallet | forgot password? | set up your password | next | click"
     And I click "Next" button during wallet setup
     Then I validate latest analytics single event "unlock wallet | forgot password? | recovery phrase length | next | click"
-    And I am on "Mnemonic verification" last page from "Forgot password" and filled all words
```
Comment (ljagiela) [inline packages/e2e-tests/src/features/analytics/AnalyticsForgotPassword.feature:53]: is it expected?

File: packages/e2e-tests/src/elements/MainLoader.ts [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 
-  get image() {
-    return $(this.MAIN_LOADER_IMAGE);
+  get mainLoaderText() {
+    return $(this.MAIN_LOADER_TEXT);
   }
 
-  get text() {
-    return $(this.MAIN_LOADER_TEXT);
+  async waitUntilLoaderDisappears() {
```
Comment (ljagiela) [inline packages/e2e-tests/src/elements/MainLoader.ts:13]: I like it!

**Review by ljagiela — CHANGES_REQUESTED** [review]

File: packages/e2e-tests/src/elements/onboarding/mainPage.ts [hunk 3]
```diff
@@ -91,6 +95,33 @@ class OnboardingMainPage extends CommonOnboardingElements {
...
+        break;
+      case 'Terms of service':
+        isMainPage ? await this.agreementTermsOfServiceLink.click() : await this.termsOfServiceLink.click();
+        break;
+      default:
+        throw new Error(`Unsupported legal link - ${link}`);
+    }
+  }
+
+  async restoreWallet(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (emiride) [reply packages/e2e-tests/src/elements/onboarding/mainPage.ts:115]: I am completely fine with , that, it was part of onboardingPageObject and when trying to get rid of it, I moved it here. I will move it to steps.

File: packages/e2e-tests/src/features/analytics/AnalyticsForgotPassword.feature (hunk 1 above)
Comment (emiride) [reply packages/e2e-tests/src/features/analytics/AnalyticsForgotPassword.feature:53]: This whole feature file is pending, so steps here will need to be changed altogether. During the refactoring this steps was removed from all other places, but only this one was left. To preserve what step was there I commented it out. @oldGreg5 is currently working on these analytics tests.

File: packages/e2e-tests/src/elements/onboarding/mainPage.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (emiride) [reply packages/e2e-tests/src/elements/onboarding/mainPage.ts]: OK in the meantime I actually did that, I just forgot to remove this method. Thanks for noticing.

File: packages/e2e-tests/src/elements/onboarding/mainPage.ts [hunk 4]
```diff
@@ -91,6 +95,33 @@ class OnboardingMainPage extends CommonOnboardingElements {
   get agreementPrivacyPolicyLink() {
     return $(this.AGREEMENT_PRIVACY_POLICY_LINK);
   }
+
+  async clickOnLegalLink(link: string, isMainPage = false): Promise<void> {
```
[the remark this answers is not in the document]
Comment (emiride) [reply packages/e2e-tests/src/elements/onboarding/mainPage.ts]: resolved

**Review by ljagiela — APPROVED** [review]

**Comment by wklos-iohk** [comment]
Can we remove `OnboardingLegalPage.ts` and relevant usages of this page? It doesn't seem to be needed anymore

File: packages/e2e-tests/src/pageobject/onboardingPageObject.ts
Comment (wklos-iohk) [inline packages/e2e-tests/src/pageobject/onboardingPageObject.ts:1]: Can you please try to move all necessary bits of code to respective POMs and get rid of this file?

File: packages/e2e-tests/src/steps/onboardingSteps.ts [hunk 5]
```diff
@@ -41,7 +38,7 @@ const twelveMnemonicWords: string[] = getTestWallet(TestWalletName.TwelveWordsMn
 const fifteenMnemonicWords: string[] = getTestWallet(TestWalletName.FifteenWordsMnemonic).mnemonic ?? [];
 
 const mnemonicWordsForReference: string[] = [];
-const validPassword = 'N_8J@bne87A';
+// const validPassword = 'N_8J@bne87A';
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/onboardingSteps.ts:41]: Is this needed?

File: packages/e2e-tests/src/steps/onboardingSteps.ts [hunk 6]
```diff
@@ -392,7 +306,13 @@ When(/^I click "Help and support" button during wallet setup$/, async () => {
 });
 
 Given(/^I restore a wallet$/, async () => {
-  await OnboardingPageObject.restoreWallet();
+  await OnboardingMainPage.restoreWalletButton.click();
+  await OnboardingWalletSetupPage.goToWalletSetupPage(
+    'Restore',
+    getTestWallet(TestWalletName.TestAutomationWallet).mnemonic ?? []
+  );
+  OnboardingWalletSetupPage.clickEnterWalletButton();
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/onboardingSteps.ts:314]: Missing `await` keyword

**Review by wklos-iohk — CHANGES_REQUESTED** [review]

File: packages/e2e-tests/src/elements/onboarding/mainPage.ts [hunk 7]
```diff
@@ -91,6 +91,22 @@ class OnboardingMainPage extends CommonOnboardingElements {
   get agreementPrivacyPolicyLink() {
     return $(this.AGREEMENT_PRIVACY_POLICY_LINK);
   }
+
+  async clickOnLegalLink(linkText: string, isMainPage = false): Promise<void> {
```
[the remark this answers is not in the document]
Comment (emiride) [reply packages/e2e-tests/src/elements/onboarding/mainPage.ts:99]: Completely agree, thanks for this

File: packages/e2e-tests/src/elements/onboarding/recoveryPhrasePage.ts [hunk 8]
```diff
@@ -72,6 +80,35 @@ class RecoveryPhrasePage extends CommonOnboardingElements {
     }
   }
 
+  async clickOnInput() {
+    const inputs = await this.mnemonicInputs;
+    await inputs[0].click();
+  }
+
+  async clickHeaderToLooseFocus() {
```
[the remark this answers is not in the document]
Comment (emiride) [reply packages/e2e-tests/src/elements/onboarding/recoveryPhrasePage.ts]: fixed

File: packages/e2e-tests/src/elements/MainLoader.ts [hunk 9]
```diff
@@ -1,18 +1,20 @@
...
+  private MAIN_LOADER_TEXT = '[data-testid="main-loader-text"]';
 
-  get component() {
+  get mainLoaderComponent() {
     return $(this.MAIN_LOADER_COMPONENT);
   }
 
-  get image() {
-    return $(this.MAIN_LOADER_IMAGE);
+  get mainLoaderText() {
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/elements/MainLoader.ts:9]: seems to be unused

File: packages/e2e-tests/src/features/OnboardingRestoreWallet.feature [hunk 10]
```diff
@@ -160,7 +160,7 @@ Feature: Onboarding - Restore wallet
     When I click "Back" button during wallet setup
     Then "Get started" page is displayed
 
-  @LW-6080 @LW-5839 @LW-5838 @LW-5839
+  @LW-6080 @LW-5839 @LW-5838
   Scenario Outline: Restore Wallet - "Recovery phrase length page" restore <mnemonicLength> words happy path
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/features/OnboardingRestoreWallet.feature:164]: We might need to update scenario titles, as there is no such page like `Recovery phrase length page`

File: packages/e2e-tests/src/steps/multidelegationSteps.ts [hunk 11]
```diff
@@ -29,9 +29,10 @@ import PortfolioBarAssert from '../assert/multidelegation/PortfolioBarAssert';
 import ChangingStakingPreferencesModalAssert from '../assert/multidelegation/ChangingStakingPreferencesModalAssert';
 import { StakePoolListColumnType, StakePoolSortingOptionType } from '../types/staking';
 import SwitchingStakePoolModal from '../elements/staking/SwitchingStakePoolModal';
-import OnboardingPageObject from '../pageobject/onboardingPageObject';
 import MoreOptionsComponentAssert from '../assert/multidelegation/MoreOptionsComponentAssert';
 
+const validPassword = 'N_8J@bne87A';
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/multidelegationSteps.ts:34]: This variable has only one usage

**Review by wklos-iohk — APPROVED** [review]
LGTM!

Minor comments added, you may handle them in separate PR.

**Pull request merged**